std/math.d

```diff
@@ -7607,3 +7607,35 @@ T truncPow2(T)(const T val) if (isFloatingPoint!T)
         assert(truncPow2(T.init).isNaN);
     }
 }
+
+// Credit: Matthias Bentrup
+/**
+Params:
+    x = the number to test
+
+Returns:
+    `true` if `x` is a nonzero power of two.
+*/
+@safe @nogc nothrow pure
+bool isPowerOf2(size_t x)
```

Comment:
```D
bool isPowerOf2(X)(X x)
if (isIntegral!X)
{}
```

On a 32 bit platform a user may want to test a `long` but don't use `long` directly because depending on the compiler the value may be passed in two registers, always (when no inlining), after convertion to `long`. With type inference this will only happen when necessary.